src/sentry/static/sentry/app/components/events/interfaces/stacktraceLinkModal.tsx

```diff
@@ -0,0 +1,207 @@
+
+  onHandleChange(sourceCodeInput: string) {
+    this.setState({
+      sourceCodeInput,
+    });
+  }
+
+  handleSubmit = async () => {
+    const {sourceCodeInput} = this.state;
+    const {organization, filename, project} = this.props;
+    const endpoint = `/projects/${organization.slug}/${project.slug}/repo-path-parsing/`;
+    try {
+      const configData = await this.api.requestPromise(endpoint, {
+        method: 'POST',
+        data: {
+          sourceUrl: sourceCodeInput,
+          stackPath: filename,
+        },
+      });
+      this.createConfig(configData);
+    } catch (err) {
+      const error = err?.responseJSON?.sourceUrl;
+      if (error && error.length > 0) {
+        addErrorMessage(t(`${error[0]}`));
+      } else {
+        addErrorMessage(t('An error occurred'));
+      }
+    }
+  };
+
+  renderBody() {
+    const {showModal, sourceCodeInput} = this.state;
+    const {filename, integrations, organization} = this.props;
+    const baseUrl = `/settings/${organization.slug}/integrations`;
+
+    return (
+      <CodeMappingButtonContainer columnQuantity={2}>
+        {t('Enable source code stack trace linking by setting up a code mapping.')}
+        <Button onClick={() => this.openModal()} size="xsmall">
+          {t('Setup Code Mapping')}
+        </Button>
+        <Modal
+          show={showModal}
+          onHide={() => this.closeModal()}
+          enforceFocus={false}
+          backdrop="static"
+          animation={false}
+        >
+          <Modal.Header closeButton>{t('Code Mapping Setup')}</Modal.Header>
+          <Modal.Body>
+            <ModalContainer>
+              <div>
+                <h6>{t('Quick Setup')}</h6>
+                {tct(
+                  'Enter in your source code url that corresponsds to stack trace filename [filename]. We will create a code mapping with this information.',
+                  {
+                    filename: <code>{filename}</code>,
+                  }
+                )}
+              </div>
+              <SourceCodeInput>
+                <StyledInputField
+                  inline={false}
+                  flexibleControlStateSize
+                  stacked
+                  name="source-code-input"
+                  type="text"
+                  value={sourceCodeInput}
+                  onChange={val => this.onHandleChange(val)}
+                  placeholder={t(
+                    `https://github.com/helloworld/Hello-World/blob/master/${filename}`
+                  )}
+                />
+                <ButtonBar>
+                  <Button type="button" onClick={() => this.handleSubmit()}>
+                    {t('Submit')}
+                  </Button>
+                </ButtonBar>
+              </SourceCodeInput>
+              <div>
+                <h6>{t('Manual Setup')}</h6>
+                {t(
+                  'To set up a code mapping manually, select which of your following integrations you want to set up the mapping for:'
```

Comment:
@MeredithAnya maybe we can add the blurb about having multiple repos here

Reply:
yeah def where I wanna add that